src/core/ruby_version.rs

```diff
@@ -664,11 +664,95 @@ macro_rules! get_ruby_string_array_3_2_0(
             // the reason I am not checking is that I don't know how to check yet
             let path_addr: usize = unsafe { rarray.as_.ary[0] as usize }; // 0 => relative path
             let abs_path_addr: usize = unsafe { rarray.as_.ary[1] as usize }; // 1 => absolute path
+
             let rel_path = get_ruby_string(path_addr, source)?;
             // In the case of internal ruby functions (and maybe others), we may not get a valid
             // pointer here
             let abs_path = get_ruby_string(abs_path_addr, source)
                 .unwrap_or(String::from("unknown"));
+
+            Ok((rel_path, abs_path))
+        }
+    )
+);
+
+macro_rules! get_ruby_string_array_3_3_0(
```

Comment:
Needed to add these as I was getting a string allocation error when trying to read the absolute path in my coredump, rstring was reporting its size as enormous (might have been reported as max string length, or max string length -1?)

I added a sanity test where we can read the value only if it doesn't exceed a maximum size, and then pass 4096 as that is the maximum path size on linux.